ha/core/system_health/system_health.py

```diff
@@ -402,41 +427,72 @@ def process_event(self, healthevent: HealthEvent):
             component_id = healthevent.resource_id
             Log.info(f"SystemHealth: Processing {component}:{component_type}:{component_id} with status {status}")
 
+            # Update the node map
+            self.node_id = healthevent.node_id
+            self.node_map = {'cluster_id':healthevent.cluster_id, 'site_id':healthevent.site_id,
+                             'rack_id':healthevent.rack_id, 'storageset_id':healthevent.storageset_id}
+
             # Read the currently stored health value
             current_health = self.get_status_raw(component, component_id, comp_type=component_type,
                                         cluster_id=healthevent.cluster_id, site_id=healthevent.site_id,
                                         rack_id=healthevent.rack_id, storageset_id=healthevent.storageset_id,
                                         node_id=healthevent.node_id, server_id=healthevent.node_id,
                                         storage_id=healthevent.node_id)
 
+            current_timestamp = str(int(time.time()))
             if current_health:
+                current_health_dict = json.loads(current_health)
+                specific_info = current_health_dict["events"][0]["specific_info"]
+            if current_health and specific_info:
+                # If health is already stored and its a node_health, check further
+                stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
+                incoming_generation_id = healthevent.specific_info["generation_id"]
+                incoming_health_status = current_health_dict["events"][0]["status"]
+                pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                 # Update the current health value itself.
-                updated_health = EntityHealth.read(current_health)
+                latest_health = EntityHealth.read(current_health)
+                if stored_genration_id != incoming_generation_id:
+                    if incoming_health_status == status:
```

Comment:
Do we need a check here as status == online also ? 
will there be any situation when stored_machine_id = incoming machine_is AND generation id != incoming_generation_id and stored_status = incoming_status = failed ?

Reply:
I think No:
POD: 123456
gen_id: cortx-data-456

case 1: POD failure (normal. Not because of deletion).
Its normal pod failure(not a delete). Hence, sys health will be stored as:
site/<site_id>/cluster/<cluster_id>/rack/<rack_id>/node/123456/health: {"specific_info": {"gen_id": "cortx-data-456", "pod_restart": 0}}
So, in that case, POD will try to restart this POD and after successful start, it comes up with same gen_id. (My third test case shows that). Incase it is failing multiple times, still the generation_id will be same. Hence, this if will be skipped.

Now, if after that, because of multiple failures, k8s decides to delete it and start new one, then according to observation we got earlier, online will come first and then failed. And so, this if will be executed properly OR that POD will remain in failed state.
So, I don't think, this case we will come across.